### PR TITLE
MM-1946 Migrate tests from "web/web_test.go" to use testify #12794

### LINKED ISSUE #12794: Migrate tests from "web/web_test.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `web/web_test.go` file.

The expected way to migrate this is to go to the `web/web_test.go` file and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in <https://community.mattermost.com/>

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19461

### PR DESCRIPTION
**Pull request by sascha-andres**
Summary:

Migrate tests from "web/web_test.go" to use testify

Ticket Link

[MM-1946](https://mattermost.atlassian.net/browse/MM-19461)

Fixes #12794

### CODE REVIEW

**Comment by sascha-andres** [comment]
@hanzei the fail comes from some frontend stuff. How does this PR proceed now?

**Comment by hmhealey** [comment]
I'll restart the tests. I think those issues have been fixed now

File: web/web_test.go [hunk 1]
```diff
@@ -253,10 +250,7 @@ func TestCheckClientCompatability(t *testing.T) {
 		{"Safari Mobile", "Mozilla/5.0 (iPhone; CPU iPhone OS 9_1 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13B137 Safari/601.1", true},
 	}
 	for _, browser := range uaTestParameters {
-		t.Run(browser.Name, func(t *testing.T) {
-			if result := CheckClientCompatability(browser.UserAgent); result != browser.Result {
-				t.Fatalf("%s User Agent Test failed!", browser.Name)
-			}
-		})
+		result := CheckClientCompatability(browser.UserAgent)
```
Comment (hmhealey) [inline web/web_test.go]: Can you re-add the `t.Run` around these so that the different test cases still show up as their own subtests?

Comment (sascha-andres) [reply web/web_test.go]: done

File: web/web_test.go [hunk 2]
```diff
@@ -254,9 +251,8 @@ func TestCheckClientCompatability(t *testing.T) {
 	}
 	for _, browser := range uaTestParameters {
 		t.Run(browser.Name, func(t *testing.T) {
-			if result := CheckClientCompatability(browser.UserAgent); result != browser.Result {
-				t.Fatalf("%s User Agent Test failed!", browser.Name)
-			}
+		  result := CheckClientCompatability(browser.UserAgent)
```
Comment (hmhealey) [inline web/web_test.go]: Thanks for re-adding that. Unfortunately, that messed up the indentation here by mixing spaces and tabs. Would you be able to fix that as well?
```suggestion
			result := CheckClientCompatability(browser.UserAgent)
```

**Comment by hmhealey** [comment]
Thanks for putting up with my nitpicking over styling 😛. I'll just merge master into this to ensure there's no issues, and then we can merge this back into master

**Comment by hmhealey** [comment]
Actually, it seems like I don't have permission to do that. Could you merge the latest master into this so that we can run that final round of tests?

**Comment by sascha-andres** [comment]
@hmhealey done

**Comment by hmhealey** [comment]
Thanks!

**Pull request merged**